pilot_experiment/2_explore_corr_of_genes.ipynb

```diff
@@ -1397,188 +1405,293 @@
+    "\n",
+    "# Distribution plot for core genes\n",
+    "sns.distplot(real_core_corr_score, \n",
+    "             label='core', \n",
+    "             color='red',\n",
+    "             bins=bins_corr,\n",
+    "             kde=False,\n",
+    "             ax=axes[0,0]\n",
+    "            )\n",
+    "\n",
+    "sns.distplot(real_acc_corr_score,\n",
+    "             label='accessory',\n",
+    "             color='blue',\n",
+    "             bins=bins_corr,\n",
+    "             kde=False,\n",
+    "             ax=axes[0,1]\n",
+    "            )\n",
+    "sns.distplot(real_core_acc_corr_score, \n",
+    "             label='core-accessory', \n",
+    "             color='purple',\n",
+    "             bins=bins_corr,\n",
+    "             kde=False,\n",
+    "             ax=axes[1,0]\n",
+    "            )\n",
+    "sns.distplot(shuffled_all_corr_score,\n",
+    "             label='shuffled', \n",
+    "             color='grey',\n",
+    "             bins=bins_corr,\n",
+    "             kde=False,\n",
+    "             ax=axes[1,1]\n",
+    "            )\n",
+    "\n",
+    "plt.suptitle('Histogram of correlation scores per group')\n",
+    "axes[0,0].set_title('Core-Core')\n",
+    "axes[0,1].set_title('Accessory-Accessory')\n",
+    "axes[1,0].set_title('Core-Accessory')\n",
+    "axes[1,1].set_title('Shuffled')\n",
+    "fig.text(0.5, 0.01, 'Correlation between genes', ha='center', fontsize=12)\n",
+    "fig.text(0.01, 0.5, 'Count', ha='center', rotation=90, fontsize=12)\n",
+    "plt.tight_layout(pad=0.4, \n",
+    "                 w_pad=0.5,\n",
+    "                 h_pad=1.0,\n",
+    "                 rect=[0, 0.03, 1, 0.95])"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "**Some sanity checks**\n",
+    "**Note about visualizations:**\n",
+    "* Based on the density plot, we observed a shift in the accessory-accessory gene correlation scores. This density plot represents the probability of a random variable falling within a particular range of values (P(0 <= X <= 0.5)). This probability is given by the integral of this variable's PDF over that range -- that is it is given by the area under the density function. So there is a higher likelihood of correlation scores > 0.5 for accessory-accessory genes compared to core-core genes. But the *exact probability* of accessory-accessory genes having a correlation score > 0.5 is not known from just visually inspecting the density plot. To get the exact probability we would need to calculate the integral from 0.5 onward.\n",
+    "* While the shift if very clear to see in the density plot, the meaning of the y-axis is not as intuitive, so we also plot histograms for each group of genes. The histogram also shows a shift in the accessory-accessory gene correlation scores and also shows that the number of accessory-accessory interactions is orders of magnitude lower compared to core-core interactions.\n",
```

Comment:
To me, if we calculate the skewness of each distribution, this will give a quantification for how different the correlation distributions are from the shuffled dataset. So it will quantify that we have some signal in our data. But we're mainly interested in the shift between the accessory-accessory (blue) compared to the other distributions.

If I run the skewtest i get the following results
```
core-core skewness:  SkewtestResult(statistic=22.532166392187893, pvalue=2.008873119085481e-112)
acc-core skewness:  SkewtestResult(statistic=-14.491763183083687, pvalue=1.3659105195229116e-47)
acc-acc skewness:  SkewtestResult(statistic=78.40894732416196, pvalue=0.0)
shuffled skewness:  SkewtestResult(statistic=175.67273941219565, pvalue=0.0)
```
I'm not sure why core-core and shuffled are coming back with significant skewness.